cpp/bench/neighbors/knn.cuh

```diff
@@ -26,15 +26,12 @@
 
 #if defined RAFT_DISTANCE_COMPILED
 #include <raft/distance/specializations.cuh>
-#include <raft/neighbors/specializations/ivf_pq_build.cuh>
+#include <raft/neighbors/specializations/ivf_pq.cuh>
 #endif
 
 #if defined RAFT_NN_COMPILED
-#include <raft/spatial/knn/specializations.cuh>
-#if defined RAFT_DISTANCE_COMPILED
-#include <raft/cluster/specializations.cuh>
```

Comment:
I had included the clustering specializations here specifically because we are using kmeans within ivfpq. Maybe this is unneeded? Now that the two kmeans impls are sharing a bunch of their core, unless we are instantiating something in ivfpq up the stack from kmeans, we should probably still include this (though it's probably better in the actual knn specializations header if it is actually needed).

Reply:
Ah, I forgot that the balanced k-means now lives in these headers as well. Do we already have specializations for it?

Reply:
I could be mistaken but I believe @Nyrio instantiated the fused 1nn because it was the prime bottleneck in kmeans. Whether or not that instantiation propagates to the balanced algorithm, im not sure.